ruby-training/trainees/david/test_katas/08/lib/playing_card_deck.rb

```diff
@@ -4,10 +4,23 @@
 class PlayingCardDeck < Deck
   def initialize
     super
-    # your code here
+    @ranks = [*(2..10), 'J', 'Q', 'K', 'A']
+    @suits = ['clubs', 'hearts', 'spades', 'diamonds']
+ 
+    @ranks.each do |rank|
+      @suits.each do |suit|
+        @cards << PlayingCard.new(rank,suit)
+      end
+    end
+
+    shuffle
   end
 
   def deal_poker_hands(count)
```

Comment:
At first glance you can see this method is doing too much. There's a `map` with a lambda over a range and then an `each` block over another range and finally a nested `each` block that populates the empty arrays that were generated a couple lines before. Whoa... Just looking at that method would make a newbie switch to python before giving ruby a try hahaha. I sincerely think you'll have a lot of fun refactoring this one!